### PR TITLE
More more builtins

### PR DESCRIPTION
**Pull request by LemonBoy**
Another day, another dump.

### CODE REVIEW

**Comment by andrewrk** [comment]
Nice! I think this is most of the work to adding `__floatsitf` as well, which would check off the next todo item on  #2443.

**Comment by LemonBoy** [comment]
> I think this is most of the work to adding __floatsitf as well

Done ;)

File: std/special/compiler_rt/floatdidf_test.zig [hunk 1]
```diff
@@ -0,0 +1,53 @@
+const __floatdidf = @import("floatdidf.zig").__floatdidf;
+const testing = @import("std").testing;
+
+fn test__floatdidf(a: i64, expected: f64) void {
+    const r = __floatdidf(a);
+    testing.expect(r == expected);
+}
+
+test "floatdidf" {
```
Comment (daurnimator) [inline std/special/compiler_rt/floatdidf_test.zig]: Why is this test in a separate file? While e.g. `floatsiXf` has tests in the same file?

Comment (andrewrk) [reply std/special/compiler_rt/floatdidf_test.zig]: There's a precedent for having them in a separate file to save space of the shipped .zig files. However it's not required.

Comment (daurnimator) [reply std/special/compiler_rt/floatdidf_test.zig]: I don't mind either way, I just think it should be consistent.

Comment (LemonBoy) [reply std/special/compiler_rt/floatdidf_test.zig]: I try to put long test cases in a separate file and keep short ones at the bottom of the file where the implementation is.

**Pull request merged**